core/src/main/scala/kafka/server/AbstractFetcherManager.scala

```diff
@@ -162,6 +162,9 @@ abstract class AbstractFetcherManager[T <: AbstractFetcherThread](val name: Stri
 
         fetcherThread.addPartitions(initialOffsetAndEpochs)
         info(s"Added fetcher to broker ${brokerAndFetcherId.broker} for partitions $initialOffsetAndEpochs")
+
+        failedPartitions.removeAll(partitionAndOffsets.keySet)
+        info(s"Removed from failed partitions set ${partitionAndOffsets.keySet}")
```

Comment:
This message is a little misleading if the partitions were not contained in `failedPartitions`. Maybe we can leave it out?